### ISSUE TITLE
File name collision due to unversioned ruby shared library

### ISSUE THREAD

**Issue by oysstu** [gazebosim/gz-math#638]
The ruby shared library is unversioned (i.e., /usr/lib/ruby/gz/math.so). Different major releases of gz-math therefore has a file with the same name and cannot be installed concurrently. Adding a major version to the shared library would fix this.
https://github.com/gazebosim/gz-math/blob/eeb17eae96b027af9497b099f706abf0637f96ec/src/ruby/CMakeLists.txt#L45C7-L45C25

I'm not a ruby user, so I can't properly test the implications of making the change.

**Comment by scpeters** [comment]
thanks for the bug report

> Adding a major version to the shared library would fix this.

Yes, adding a major version to the shared library is one options but I believe this would also mean that the `require` (like `import`) statements would need the major version as well (see [angle_example.rb](https://github.com/gazebosim/gz-math/blob/gz-math8/examples/angle_example.rb#L23))

I think we should do the following for our released packages:

* [ ] Add a conflict in the Debian metadata for the `ruby-gz-math*` packages (see [gz-math8-release control file](https://github.com/gazebo-release/gz-math8-release/blob/main/ubuntu/debian/control#L92-L109) for example)
* [ ] Add a caveat about the name collision / conflict for ruby bindings to the [build-from-source documentation](https://github.com/gazebosim/gz-math/blob/gz-math8/tutorials/installation.md?plain=1#L109-L112)

**Comment by azeey** [comment]
I agree with @scpeters. Since the `ruby-gz-math*` packages are not part of the metapackages (e.g. `gz-harmonic`, `gz-ionic`), adding the conflict won't be a problem. This would mean that it will not be possible to install the Ruby bindings side by side, which is the case currently as pointed to by this issue, but this would make it official.
